activesupport/lib/active_support/core_ext/object/json.rb

```diff
@@ -65,6 +65,14 @@ def as_json(options = nil) # :nodoc:
   end
 end
 
+if RUBY_VERSION >= "3.2"
+  class Data # :nodoc:
+    def as_json(options = nil)
+      Hash[members.zip(deconstruct)].as_json(options)
```

Comment:
Mainly for education purposes, can we implement it as 
```suggestion
      to_h.as_json(options)
```
which feels like a higher-level approach that is simpler to maintain. It looks like `Struct` uses a similar `zip` approach but I couldn't find a specific reason for that

Reply:
`to_h` is quicker so perhaps a better option unless I'm missing the difference
```ruby
Benchmark.ips do |x|
  x.report("to_h") { data_obj.to_h }
  x.report("members.zip(deconstruct)") { Hash[data_obj.members.zip(data_obj.deconstruct)] }

  x.compare!
end
```

```
Calculating -------------------------------------
                to_h      3.026M (±15.8%) i/s -     14.943M in   5.063868s
members.zip(deconstruct)
                          1.304M (±20.9%) i/s -      6.287M in   5.047154s

Comparison:
                to_h:  3025508.3 i/s
members.zip(deconstruct):  1304149.5 i/s - 2.32x  (± 0.00) slower
```

Reply:
@nvasilevski out of curiosity, can you run the same benchmark for Struct?

Reply:
Looks to me that `Array#zip` was used for historical reasons, before Struct supported `to_h`.

I've updated the code and linked the benchmarks as well: https://github.com/rails/rails/commit/a7d4d78479fc81f94ccd5e5a505f708ec87bf318

@nvasilevski thanks for your help!